### PR TITLE
Identify more bit flags

### PR DESCRIPTION
**Pull request by Rangi42**
Pretty self-explanatory; a similar effort to some of the recent pokered commits.

Some things to note:

- I did this in pokegold, not pokecrystal, so the unidentified mobile stuff wouldn't be a distraction. This will need porting to there once it's approved.
- There are many `bit/res/set 7`s left, because of a convention to set the high bit of jumptable indexes to mean "exit the jumptable". I may add a generic `EXIT_JUMPTABLE_F` constant in a separate PR.
- I edited the `shift_const` macro to automatically define `\1` (bit mask) and `\1_F` (0-7 bit flag) constants. This does make it harder to grep for <code><i>NAME</i>_F</code> definitions, but otherwise we'd have a lot of redundant definition lines.
- We don't have a perfect convention of naming flag constants with `_F` or not. I usually went for local consistency, i.e. imitate nearby constant names.
- A few of these are surprising/weird, like how `GetMovementPermissions` in home/map.asm sets the `RIGHT` bit in all four directional cases. I may have made mistakes there, but AFAICT that's what the bit means.

### CODE REVIEW

File: constants/ram_constants.asm [hunk 1]
```diff
@@ -118,21 +125,23 @@ DEF INIT_OTHER_ITEM_LIST EQU 3
 DEF INIT_PLAYEROT_LIST   EQU 4
 DEF INIT_MON_LIST        EQU 5
 
+; wEnteredMapFromContinue::
+DEF ENTERED_MAP_FROM_CONTINUE_2 EQU 2
+DEF ENTERED_MAP_FROM_CONTINUE_7 EQU 7
```
Comment (Rangi42) [inline constants/ram_constants.asm:130]: I'm pretty sure these are unused.

File: constants/ram_constants.asm [hunk 2]
```diff
@@ -163,7 +176,20 @@ DEF PLAYERSPRITESETUP_RESET_ACTION_F    EQU 7
 	const MAPEVENTS_OFF ; 1
 
 ; wScriptFlags::
-DEF SCRIPT_RUNNING EQU 2
+	const_def
+	const SCRIPT_FLAG_0
+	const SCRIPT_FLAG_1
```
Comment (Rangi42) [inline constants/ram_constants.asm:181]: These look unused.

File: constants/ram_constants.asm [hunk 3]
```diff
@@ -163,7 +176,20 @@ DEF PLAYERSPRITESETUP_RESET_ACTION_F    EQU 7
...
+	const RUN_DEFERRED_SCRIPT
+
+; wEnabledPlayerEvents::
+	const_def
+	const PLAYEREVENTS_COUNT_STEPS
+	const PLAYEREVENTS_COORD_EVENTS
+	const PLAYEREVENTS_WARPS_AND_CONNECTIONS
+	const_skip
+	const PLAYEREVENTS_WILD_ENCOUNTERS
+	const PLAYEREVENTS_5
```
Comment (Rangi42) [inline constants/ram_constants.asm:192]: Probably unused but might be set somewhere by an `or`.

File: engine/battle_anims/functions.asm [hunk 4]
```diff
@@ -807,6 +807,9 @@ BattleAnimFunction_FireBlast:
 	ret
 
 BattleAnimFunction_RazorLeaf:
+; Object moves at an arc
+; Obj Param: Bit 6 defines offset from base frameset BATTLE_ANIM_FRAMESET_RAZOR_LEAF_2
+;            Rest defines arc radius
```
Comment (Rangi42) [inline engine/battle_anims/functions.asm:812]: Similar code to Rock Smash below it.

File: engine/overworld/player_movement.asm [hunk 5]
```diff
@@ -665,7 +665,7 @@ ENDM
 
 	ld hl, OBJECT_FLAGS2
 	add hl, bc
-	set 2, [hl]
+	set OBJ_FLAGS2_2, [hl]
```
Comment (Rangi42) [inline engine/overworld/player_movement.asm:668]: Not sure what this is for.

File: home/map.asm [hunk 6]
```diff
@@ -1957,7 +1957,7 @@ GetMovementPermissions::
 
 .ok_down
 	ld hl, wTilePermissions
-	set 3, [hl]
+	set RIGHT, [hl]
```
Comment (Rangi42) [inline home/map.asm:1960]: I don't understand why it's the `RIGHT` bit in all four directions, but it really seems that way. Inconsequential copy+paste bug?

File: engine/overworld/player_movement.asm (hunk 5 above)
Comment (vulcandth) [reply engine/overworld/player_movement.asm:668]: Looks to me like a boulder moving flag... 

Its set here if the object the player is running into is a boulder.
Its checked in MovementFunction_Strength to determine if the function should move the boulder or not.
its reset in the StepFunction_StrengthBoulder when the movement is complete.

So maybe... `BOULDER_MOVING_F`?

File: ram/wram.asm [hunk 7]
```diff
@@ -2212,6 +2213,7 @@ wMapStatus:: db
 wMapEventStatus:: db
 
 wScriptFlags::
+; bit 2: running script
 ; bit 3: run deferred script
```
Comment (vulcandth) [inline ram/wram.asm:2217]: Once we define constants for the bits.. we don't really need the comments here anymore imo.

File: constants/ram_constants.asm (hunk 1 above)
Comment (vulcandth) [reply constants/ram_constants.asm:130]: I think so as well; but I can double check in game with a few tests to see if I can catch something might have missed.

File: ram/wram.asm (hunk 7 above)
Comment (Rangi42) [reply ram/wram.asm:2217]: Maybe not, but we already have a lot of bit comments when constants already exist, so I didn't want to delete them all in this PR.

File: constants/ram_constants.asm (hunk 2 above)
Comment (vulcandth) [reply constants/ram_constants.asm:181]: I concur that they are unused.

File: constants/ram_constants.asm (hunk 1 above)
Comment (Rangi42) [reply constants/ram_constants.asm:130]: If so, `wEnteredMapFromContinue` should be renamed, perhaps to `wUnusedMapEntryFlags`.

File: constants/ram_constants.asm (hunk 2 above)
Comment (Rangi42) [reply constants/ram_constants.asm:181]: `UNUSED_SCRIPT_FLAG_0/1`, then.

File: home/map.asm (hunk 6 above)
Comment (vulcandth) [reply home/map.asm:1960]: Yeah; this looks like a copy+paste bug. It appears to have been fixed/changed in pokecrystal.

File: constants/menu_constants.asm [hunk 8]
```diff
@@ -37,6 +36,22 @@
...
+	shift_const _2DMENU_EXIT_DOWN
+	shift_const _2DMENU_WRAP_LEFT_RIGHT
+	shift_const _2DMENU_WRAP_UP_DOWN
+	shift_const _2DMENU_ENABLE_SPRITE_ANIMS
+	shift_const _2DMENU_DISABLE_JOYPAD_FILTER
+
+; _2DMenu more flags 2
+	const_def
+	const_skip 7
+	shift_const _2DMENU_EXITING
```
Comment (vulcandth) [inline constants/menu_constants.asm:53]: Lets add bit index comments here.

File: constants/printer_constants.asm [hunk 9]
```diff
@@ -7,3 +7,14 @@
...
+; wPrinterStatusFlags
+	const_def 5
+	shift_const PRINTER_STATUS_ERROR_3
+	shift_const PRINTER_STATUS_ERROR_4
+	shift_const PRINTER_STATUS_ERROR_1
+
+; wPrinterConnectionOpen
+	const_def
+	const PRINTER_CONNECTION_OPEN
+	const PRINTER_CONNECTION_SUCCESS
```
Comment (vulcandth) [inline constants/printer_constants.asm:20]: index commments

File: constants/ram_constants.asm [hunk 10]
```diff
@@ -163,7 +176,20 @@ DEF PLAYERSPRITESETUP_RESET_ACTION_F    EQU 7
 	const MAPEVENTS_OFF ; 1
 
 ; wScriptFlags::
-DEF SCRIPT_RUNNING EQU 2
+	const_def
+	const SCRIPT_FLAG_0
+	const SCRIPT_FLAG_1
+	const SCRIPT_RUNNING
+	const RUN_DEFERRED_SCRIPT
```
Comment (vulcandth) [inline constants/ram_constants.asm:183]: index comments

File: constants/ram_constants.asm (hunk 3 above)
Comment (vulcandth) [inline constants/ram_constants.asm:192]: index comments.

File: constants/ram_constants.asm [hunk 11]
```diff
@@ -300,6 +329,12 @@ DEF DAYCARELADY_HAS_MON_F        EQU 0
 ; shared flag between wDayCareMan and wDayCareLady
 DEF DAYCARE_INTRO_SEEN_F         EQU 7
 
+; sRTCStatusFlags::
+	const_def 5
+	shift_const RTC_DAYS_EXCEED_139
+	shift_const RTC_DAYS_EXCEED_255
+	shift_const RTC_RESET
```
Comment (vulcandth) [inline constants/ram_constants.asm:336]: index comments

**Review by vulcandth — CHANGES_REQUESTED** [review]
After these small changes I think we are good to merge! Thank you!

**Pull request merged**